### PR TITLE
docs: add "OWASP API Security Top 10 2023" to Security Guidelines

### PR DESCRIPTION
**Pull request by kenjis**
~~Needs #8652~~

**Description**
- add "OWASP API Security Top 10 2023"

Frankly, CI4 offers very few features for API Security.

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
This is the last PR for v4.4.8.
Review, please.

File: user_guide_src/source/concepts/security.rst [hunk 1]
```diff
@@ -578,8 +573,361 @@ Do not mitigate SSRF via the use of a deny list or regular expression. Attackers
...
+
+APIs tend to expose endpoints that handle object identifiers, creating a wide
+attack surface of Object Level Access Control issues. Object level authorization
+checks should be considered in every function that accesses a data source using
+an ID from the user.
+
+OWASP recommendations
+---------------------
+
+- Implement a proper authorization mechanism that relies on the user policies and
```
Comment (datamweb) [inline user_guide_src/source/concepts/security.rst:597]: In this case a good job is done by **lonnieezell**:
https://github.com/lonnieezell/forum-example/blob/develop/_docs/policies.md

File: user_guide_src/source/concepts/security.rst [hunk 2]
```diff
@@ -578,8 +573,361 @@ Do not mitigate SSRF via the use of a deny list or regular expression. Attackers
...
+Assignment, focusing on the root cause: the lack of or improper authorization
+validation at the object property level. This leads to information exposure or
+manipulation by unauthorized parties.
+
+OWASP recommendations
+---------------------
+
+- When exposing an object using an API endpoint, always make sure that the user
+  should have access to the object's properties you expose.
+- Avoid using generic methods such as to_json() and to_string(). Instead,
```
Comment (datamweb) [inline user_guide_src/source/concepts/security.rst:663]: There are (`to_json()` and `to_string()`)codes, it is better to specify them as codes.

**Review by datamweb — APPROVED** [review]

File: user_guide_src/source/concepts/security.rst (hunk 2 above)
Comment (kenjis) [reply user_guide_src/source/concepts/security.rst:663]: Since CI does not provide such methods, why not just leave it as it is?

**Pull request merged**